### ISSUE TITLE
Add support for right-to-left layouts / alignments

### ISSUE THREAD

**Issue by tronical** [slint-ui/slint#2294]
When Slint is running with the application translated into different languages, we may want to adjust the flow of layouts and default text alignment to accommodate right-to-left languages.

* We need API to set a layout direction (global default, per-window, or per layout).
* When text is not explicitly aligned, we should implicitly change the left alignment to right alignment (`Text` element). This might require an "Auto" value for `horizontal-alignment` that we default to.
* Provide additional API to apply a mirror effect, for example to images

**Comment by chvllad** [comment]
I think it also make sense to add `start` and `end` enum variants into [TextHorizontalAlignment](https://github.com/slint-ui/slint/blob/master/docs/language/src/builtins/enums.md#texthorizontalalignment) like in CSS [text-align](https://developer.mozilla.org/en-US/docs/Web/CSS/text-align#values).

> This might require an "Auto" value for horizontal-alignment that we default to.

I think all properties should have explicit default value akin to CSS's `unset` or `initial`. E.g., if you set `width` property to some value there seems like no way to reset it back to auto.

**Comment by tronical** [comment]
Yes, `start` and `end` might make sense for those - I like that. We already use that for the `alignment` on the horizontal and vertical text layout.

**Comment by Enyium** [comment]
> I think all properties should have explicit default value akin to CSS's `unset` or `initial`. E.g., if you set `width` property to some value there seems like no way to reset it back to auto.

This is off-topic, but if Slint's internals would be changed to accommodate this, this would also come in handy for cases where you just need to deviate from the normal value under *some* circumstances, and would prefer not to repeat and hardcode the regular value:

```slint
Text {
    text: "Bold on hover";
    font-weight: touch-area.has-hover ? 700 : initial;
}
```

`initial` could even be used like a normal value of the same type as the property:

```slint
    color: touch-area.has-hover ? initial.with-alpha(50%) : initial;
```
